### ISSUE TITLE
Incorrect merged env object from plugins file

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#7002]
Cypress 4.3.0, 4.4.0

If I set an environment value in `cypress.json` like

```json
{
  "env": {
    "foo": {}
  }
}
```

Then config shows

<img width="406" alt="Screen Shot 2020-04-13 at 11 11 43 PM" src="https://user-images.githubusercontent.com/2212006/79182199-3d1fe900-7ddc-11ea-90a0-6cb607ae0dd6.png">

If I set in `cypress.json`

```json
{
  "env": {
    "foo": {
      "bar": 42
    }
  }
}
```

Then it shows up correctly

<img width="103" alt="Screen Shot 2020-04-13 at 11 12 45 PM" src="https://user-images.githubusercontent.com/2212006/79182245-5e80d500-7ddc-11ea-8d8c-35804e87f6fc.png">

BUT if I set the same value from plugins file

```js
module.exports = (on, config) => {
  config.env.foo = {
    bar: 42,
  }
  return config
}
```
Then our TR crashes

<img width="801" alt="Screen Shot 2020-04-13 at 11 14 51 PM" src="https://user-images.githubusercontent.com/2212006/79182350-96881800-7ddc-11ea-9fa0-cbe36bc68b37.png">

Relevant part from verbose logs

```
  cypress:server:config config diffs { env: { foo: { bar: 42 } }, configFile: '/Users/gleb/git/cypress-examples/cypress.json' } +3ms
  cypress:server:config resolved config before diffs { animationDistanceThreshold: { value: 5, from: 'default' }, fileServerFolder: { value: '', from: 'default' }, baseUrl: { value: null, from: 'default' }, fixturesFolder: { value: false, from: 'config' }, blacklistHosts: { value: null, from: 'default' }, chromeWebSecurity: { value: true, from: 'default' }, modifyObstructiveCode: { value: true, from: 'default' }, integrationFolder: { value: 'docs', from: 'config' }, env: {}, pluginsFile: { value: 'plugins.js', from: 'config' }, hosts: { value: null, from: 'default' }, screenshotsFolder: { value: 'cypress/screenshots', from: 'default' }, numTestsKeptInMemory: { value: 50, from: 'default' }, supportFile: { value: false, from: 'config' }, port: { value: null, from: 'default' }, projectId: { value: null, from: 'default' }, videosFolder: { value: 'cypress/videos', from: 'default' }, reporter: { value: 'spec', from: 'default' }, reporterOptions: { value: null, from: 'default' }, ignoreTestFiles: { value: '*.hot-update.js', from: 'default' }, testFiles: { value: '**/*.md', from: 'config' }, defaultCommandTimeout: { value: 4000, from: 'default' }, trashAssetsBeforeRuns: { value: true, from: 'default' }, execTimeout: { value: 60000, from: 'default' }, userAgent: { value: null, from: 'default' }, pageLoadTimeout: { value: 60000, from: 'default' }, viewportWidth: { value: 1000, from: 'default' }, requestTimeout: { value: 5000, from: 'default' }, viewportHeight: { value: 660, from: 'default' }, responseTimeout: { value: 30000, from: 'default' }, video: { value: true, from: 'default' }, taskTimeout: { value: 60000, from: 'default' }, videoCompression: { value: 32, from: 'default' }, videoUploadOnPasses: { value: true, from: 'default' }, watchForFileChanges: { value: true, from: 'default' }, waitForAnimations: { value: true, from: 'default' }, nodeVersion: { value: 'default', from: 'default' }, firefoxGcInterval: { value: { runMode: 1, openMode: null }, from: 'default' }, browsers: { value: [ [Object], [Object], [Object], [Object], [Object], [Object], [Object], [Object], [Object], [Object] ], from: 'default' }, experimentalGetCookiesSameSite: { value: false, from: 'default' } } +0ms
  cypress:server:events send error: TypeError: Cannot set property 'bar' of undefined at /Users/gleb/Library/Caches/Cypress/4.4.0/Cypress.app/Contents/Resources/app/packages/server/lib/config.js:335:37
```

Seems there is a problem applying nested config diff from plugin

```
{ env: { foo: { bar: 42 } }
```
to the current object
```
env: {}
```

**Comment by mccataldo** [comment]
@bahmutov any workaround for this?

**Comment by bahmutov** [comment]
No, not yet, did not have time to investigate this

**Comment by jennifer-shehane** [comment]
Duplicate of #1736

**Comment by jennifer-shehane** [comment]
The code for this is done in https://github.com/cypress-io/cypress/pull/7960 and was released in [Cypress 4.11.0](https://on.cypress.io/changelog#4-11-0).

This comment thread has been locked. If you are still experiencing this issue after upgrading to
Cypress v4.11.0, please open a new issue.
